### PR TITLE
Enable exporting of specific resources in the GooglePhotosExporter and FlickrPhotosExporter

### PR DESCRIPTION
**Pull request by seehamrun**
This is a quick first pass at enabling exporting specific resources in these exporters. 

Previously they only accepted IdOnlyContainerResources in the export information. This assumes that if the ExportInformation contains a PhotosContainerResource that the export process should start there instead of listing out all the user media items

### CODE REVIEW

File: extensions/data-transfer/portability-data-transfer-flickr/src/main/java/org/datatransferproject/datatransfer/flickr/photos/FlickrPhotosExporter.java [hunk 1]
```diff
@@ -123,6 +123,14 @@ static String toMimeType(String flickrFormat) {
 
     RequestContext.getRequestContext().setAuth(auth);
 
+    if (exportInformation.isPresent()
```
Comment (holachuy) [inline extensions/data-transfer/portability-data-transfer-flickr/src/main/java/org/datatransferproject/datatransfer/flickr/photos/FlickrPhotosExporter.java:126]: Let's add a comment here to describe this logic

File: extensions/data-transfer/portability-data-transfer-flickr/src/main/java/org/datatransferproject/datatransfer/flickr/photos/FlickrPhotosExporter.java [hunk 2]
```diff
@@ -136,6 +144,37 @@ static String toMimeType(String flickrFormat) {
...
+        // Adding subresources tells the framework to recall export to get all the photos
+        subResources.add(new IdOnlyContainerResource(photoset.getId()));
+      }
+
+      for (PhotoModel photo : container.getPhotos()) {
+        Photo p = photosInterface.getInfo(photo.getDataId(), null);
+        photosBuilder.add(toCommonPhoto(p, null));
+      }
+    } catch (FlickrException e) {
+      return new ExportResult<>(e);
```
Comment (holachuy) [inline extensions/data-transfer/portability-data-transfer-flickr/src/main/java/org/datatransferproject/datatransfer/flickr/photos/FlickrPhotosExporter.java:168]: Any additional logging we want to try here?

File: extensions/data-transfer/portability-data-transfer-flickr/src/main/java/org/datatransferproject/datatransfer/flickr/photos/FlickrPhotosExporter.java [hunk 3]
```diff
@@ -136,6 +144,37 @@ static String toMimeType(String flickrFormat) {
...
+    }
+
+    PhotosContainerResource photosContainerResource =
+        new PhotosContainerResource(albumBuilder.build(), photosBuilder.build());
+    ContinuationData continuationData = new ContinuationData(null);
+    subResources.forEach(resource -> continuationData.addContainerResource(resource));
+    return new ExportResult<>(ResultType.CONTINUE, photosContainerResource, continuationData);
```
Comment (holachuy) [inline extensions/data-transfer/portability-data-transfer-flickr/src/main/java/org/datatransferproject/datatransfer/flickr/photos/FlickrPhotosExporter.java:175]: What would happen if there are no albums and no photos and each of these is empty?

File: extensions/data-transfer/portability-data-transfer-flickr/src/main/java/org/datatransferproject/datatransfer/flickr/photos/FlickrPhotosExporter.java (hunk 1 above)
Comment (seehamrun) [reply extensions/data-transfer/portability-data-transfer-flickr/src/main/java/org/datatransferproject/datatransfer/flickr/photos/FlickrPhotosExporter.java:126]: I added a comment, but it was below the if statmenet, moved it to the top

File: extensions/data-transfer/portability-data-transfer-flickr/src/main/java/org/datatransferproject/datatransfer/flickr/photos/FlickrPhotosExporter.java (hunk 2 above)
Comment (seehamrun) [reply extensions/data-transfer/portability-data-transfer-flickr/src/main/java/org/datatransferproject/datatransfer/flickr/photos/FlickrPhotosExporter.java:168]: I followed the same thing we were doing with the individual getAlbums() and getPhotos() calls. The exception would be logged up a level (in the executor)

File: extensions/data-transfer/portability-data-transfer-flickr/src/main/java/org/datatransferproject/datatransfer/flickr/photos/FlickrPhotosExporter.java (hunk 3 above)
Comment (seehamrun) [reply extensions/data-transfer/portability-data-transfer-flickr/src/main/java/org/datatransferproject/datatransfer/flickr/photos/FlickrPhotosExporter.java:175]: Then there would be nothing to import. The job will be counted as "success". 

I guess what we do here depends on what the expectation is if there is no content to export. If they specified an invalid photo/album id the lookup fails and an error gets thrown. 

If we want the job to fail, we can add a precondition check, but again not sure if thats what we want to do.

**Comment by seehamrun** [comment]
@wmorland Added you in case you were creating the ExportInformation differently.

Also note: I chose to look up the album based on the ID (though theoratically, we could take in the id, title, description and other metadata from the request) just in case between selection and the job processing the album details change or get deleted (or if theres a misconfiguration and the album doenst exist)

**Pull request merged**